core/invoker/src/main/scala/whisk/core/containerpool/Container.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.containerpool
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+import spray.json.JsObject
+import whisk.common.TransactionId
+import whisk.core.container.Interval
+import whisk.core.entity.ActivationResponse
+import whisk.core.entity.ByteSize
+
+/**
+ * An OpenWhisk biased container abstraction. This is **not only** an abstraction
+ * for different container providers, but the implementation also needs to include
+ * OpenWhisk specific behavior, especially for initialize and run.
+ */
+trait Container {
+    /** Stops the container from consuming CPU cycles. */
+    def halt()(implicit transid: TransactionId): Future[Unit]
```

Comment:
I chose a different naming than docker commands to force myself not to hardwire this with docker and to make the distinction more "clear".